src/WrappedMToken.sol

```diff
@@ -731,19 +772,26 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
 
     /**
      * @dev    Compute the adjusted amount of M that can safely be transferred out given the current index.
-     * @param  amount_       Some amount to be transferred out of this contract.
-     * @param  currentIndex_ The current index.
-     * @return safeAmount_   The adjusted amount that can safely be transferred out.
+     * @param  amount_     Some amount to be transferred out of this contract.
+     * @return safeAmount_ The adjusted amount that can safely be transferred out.
      */
-    function _getSafeTransferableM(uint240 amount_, uint128 currentIndex_) internal view returns (uint240 safeAmount_) {
```

Comment:
rationale for using such complex logical method as `_getSafeTransferableM` is not fully clear to me. Excess calculations account sfor rounding error.

Non earners on transfers will be charged exact amount of M, earners will be charged round up amount (relatively rare case) 

What use-case is exactly compromising safety and justifies introduction of `getSafeTransferableAmount`?

Reply:
`getSafeTransferableAmount` ensures at most `amount_` of M is taken from the sender, whether that be from the wrapper when claiming excess, or from the account/user when wrapping M.